src/NServiceBus.AwsLambda.SQS/TransportWrapper/ServerlessTransport.cs

```diff
@@ -1,26 +1,53 @@
 ﻿namespace NServiceBus.AwsLambda.SQS.TransportWrapper
 {
-    using Settings;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus;
     using Transport;
 
     class ServerlessTransport<TBaseTransport> : TransportDefinition
-        where TBaseTransport : TransportDefinition, new()
+        where TBaseTransport : TransportDefinition
     {
-        public ServerlessTransport()
+        // HINT: This constant is defined in NServiceBus but is not exposed
+        const string MainReceiverId = "Main";
+
+        public ServerlessTransport(TBaseTransport baseTransport)
+            : base(baseTransport.TransportTransactionMode, baseTransport.SupportsDelayedDelivery, baseTransport.SupportsPublishSubscribe, baseTransport.SupportsTTBR)
         {
-            baseTransport = new TBaseTransport();
+            BaseTransport = baseTransport;
         }
 
-        public override string ExampleConnectionStringForErrorMessage { get; } = string.Empty;
+        public TBaseTransport BaseTransport { get; }
 
-        public override bool RequiresConnectionString => baseTransport.RequiresConnectionString;
+        public PipelineInvoker PipelineInvoker { get; private set; }
 
-        public override TransportInfrastructure Initialize(SettingsHolder settings, string connectionString)
+        public override async Task<TransportInfrastructure> Initialize(HostSettings hostSettings, ReceiveSettings[] receivers, string[] sendingAddresses, CancellationToken cancellationToken = default)
         {
-            var baseTransportInfrastructure = baseTransport.Initialize(settings, connectionString);
-            return new ServerlessTransportInfrastructure<TBaseTransport>(baseTransportInfrastructure, settings);
+            //   hostSettings.
+            var baseTransportInfrastructure = await BaseTransport.Initialize(hostSettings, receivers, sendingAddresses, cancellationToken).ConfigureAwait(false);
+            var serverlessTransportInfrastructure = new ServerlessTransportInfrastructure<TBaseTransport>(baseTransportInfrastructure);
+            PipelineInvoker = (PipelineInvoker)serverlessTransportInfrastructure.Receivers[MainReceiverId];
+
+            return serverlessTransportInfrastructure;
+
         }
 
-        readonly TBaseTransport baseTransport;
+#pragma warning disable CS0672 // Member overrides obsolete member
+#pragma warning disable CS0618 // Type or memeber is obsolete
+
+        public override string ToTransportAddress(QueueAddress address) => BaseTransport.ToTransportAddress(address);
+
+#pragma warning restore CS0618 // Type or memeber is obsolete
+#pragma warning restore CS0672 // Member overrides obsolete member
+
+
+        public override IReadOnlyCollection<TransportTransactionMode> GetSupportedTransactionModes() => supportedTransactionModes;
```

Comment:
Would it be possible to delegate this forward to the base transport or is this dangerous because we might need to write explicit code in the lambda support to support a tx mode? I think it is quite unlikely though that we get SendsWithAtomicReceive with SQS / SNS

Reply:
not sure forwarding makes sense, even when the values are the same anyway. The transaction support is heavily determined by the pump which we don't control here, so the transaction modes listed here are what the trigger/hosting model allow us to use/integrate with, not what our SQS transport implementation could do.

Reply:
Agreed. My. Thought was coming from the angle that the transport pump and this pump should always be as close as possible

Reply:
Do we agree then to leave this as is?

Reply:
Good for me